src/INode.java

```diff
@@ -174,9 +175,21 @@ public ASTValue eval(WordsEnvironment environment) throws WordsRuntimeException
 		}
 	}
 
-	private ASTValue evalAdd(WordsEnvironment environment) {
-		// TODO
-		throw new AssertionError("Not yet implemented");
+	private ASTValue evalAdd(WordsEnvironment environment) throws WordsRuntimeException {
+		ASTValue lhs = children.get(0).eval(environment).tryCoerceTo(AST.ValueType.NUM);
+		ASTValue rhs = children.get(1).eval(environment).tryCoerceTo(AST.ValueType.NUM);
+		
+		if (((lhs.type == AST.ValueType.STRING) && (rhs.type == AST.ValueType.NUM)) || 
+				((lhs.type == AST.ValueType.NUM) && (rhs.type == AST.ValueType.STRING)) ||
+				((lhs.type == AST.ValueType.STRING) && (rhs.type == AST.ValueType.STRING))) {
+			lhs.tryCoerceTo(AST.ValueType.STRING);
+			rhs.tryCoerceTo(AST.ValueType.STRING);
+			return new ASTValue(lhs.stringValue+rhs.stringValue);
+		} else if ((lhs.type == AST.ValueType.NUM) && (rhs.type == AST.ValueType.NUM)) {
+			return new ASTValue(lhs.numValue+rhs.numValue); 
+		} else {
+			throw new WordsArithmeticException(lhs.type.toString(), rhs.type.toString());
```

Comment:
I think you may have forgotten to "git add" WordsArithmeticException.java, since it's not included on this branch.